rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRedundantDependencyVersionsTest.java

```diff
@@ -1818,6 +1879,65 @@ void dependencyPropertySubstitution() {
         );
     }
 
+    @Test
+    void unusedDependencyPropertiesAreRemoved() {
+        rewriteRun(
+          spec -> spec.recipe(new RemoveRedundantDependencyVersions(null, null, RemoveRedundantDependencyVersions.Comparator.GTE, null)),
+          pomXml(
+            """
+            <project>
+                <parent>
+                    <groupId>org.springframework.boot</groupId>
+                    <artifactId>spring-boot-starter-parent</artifactId>
+                    <version>2.5.4</version>
+                    <relativePath/>
+                </parent>
+
+                <groupId>com.example</groupId>
+                <artifactId>test</artifactId>
+                <version>1.0.0-SNAPSHOT</version>
+                <properties>
+                    <spring-web.version>5.3.9</spring-web.version>
+                </properties>
```

Comment:
I have some concerns about removing properties like this; in a multi module project we could remove a property from a parent that's still used in a child project, since `doAfterVisit` can only look at a single source file.

Reply:
Would a

```java
getResolutionResult().getPom().getPackaging().equals("jar")
```

[check](https://github.com/openrewrite/rewrite/pull/5645/commits/ad0551aa5bb6ecea9c8d6e54b7a874e62d4af18d) be enough?